common/unified/matrix/csr_kernels.cpp

```diff
@@ -313,6 +313,33 @@ void benchmark_lookup(std::shared_ptr<const DefaultExecutor> exec,
 GKO_INSTANTIATE_FOR_EACH_INDEX_TYPE(GKO_DECLARE_CSR_BENCHMARK_LOOKUP_KERNEL);
 
 
+template <typename ValueType, typename IndexType>
+void row_wise_absolute_sum(std::shared_ptr<const DefaultExecutor> exec,
+                           const matrix::Csr<ValueType, IndexType>* orig,
+                           array<ValueType>& sum)
+{
+    run_kernel(
+        exec,
+        [] GKO_KERNEL(auto row, auto sum_ptr) {
+            sum_ptr[row] = zero<ValueType>();
+        },
+        sum.get_num_elems(), sum.get_data());
+
+    run_kernel(
+        exec,
+        [] GKO_KERNEL(auto row, auto row_ptrs, auto value_ptr, auto sum_ptr) {
+            for (auto k = row_ptrs[row]; k < row_ptrs[row + 1]; ++k) {
+                sum_ptr[row] += abs(value_ptr[k]);
+            }
```

Comment:
```suggestion
            sum_ptr[row] = zero<device_type<ValueType>>();
            for (auto k = row_ptrs[row]; k < row_ptrs[row + 1]; ++k) {
                sum_ptr[row] += abs(value_ptr[k]);
            }
```
with this, you can delete the previous one.
need to use device_type for __half and complex